File: 02_double_dispatch/java/Paper.java
public class Paper implements Beatable {
	@Override
	public String beats(Beatable other) {
		if (other.getClass() == Rock.class) {
			return "paper wins";
		}

		if (other.getClass() == Scissor.class) {
			return "paper loses";
		}
		return "draw";
	}
}

File: 02_double_dispatch/c#/01_before_pattern_matching/Paper.cs
public class Paper : Beatable {
	public string Beats(Beatable other) {
		if (other.GetType() == typeof(Rock)) {
			return "paper wins";
		}

		if (other.GetType() == typeof(Scissor)) {
			return "paper loses";
		}
		return "draw";
	}
}
